### PR TITLE
Collect warnings and print at the end of freezing process

### LINKED ISSUE #277: Add support for warnings
Freezeyt should collect warnings like it collects errors, and present them as a list at the end.
It should also have a way to filter out warnings or treat them as errors.

### PR DESCRIPTION
**Pull request by nappex**
Fix #277

### CODE REVIEW

**Comment by nappex** [comment]
First part should be done, maybe add some tests.

I have to think about: **It should also have a way to filter out warnings or treat them as errors.** This part is not implemented yet.

**Comment by encukou** [comment]
Please also test the `Freezer.warnings` list, and the behavior with more than one warning.

**Comment by nappex** [comment]
> Please also test the `Freezer.warnings` list, and the behavior with more than one warning.

Completed :rocket:

File: tests/test_status_handlers.py [hunk 1]
```diff
@@ -52,6 +52,41 @@ def index():
     assert expected_output in captured.out
 
 
+def test_several_warns(capsys):
+    warnings = []
+
+    def record_warnings(freeze_info):
+        warnings.clear()
+        warnings.extend(freeze_info._freezer.warnings)
```
Comment (encukou) [inline tests/test_status_handlers.py:60]:
```suggestion
        warnings[:] = freeze_info._freezer.warnings
```

Comment (encukou) [inline tests/test_status_handlers.py:60]:
```suggestion
    warnings = []

    def record_warnings(task_info):
        nonlocal warnings
        warnings = list(task_info._freezer.warnings)
```

**Review by encukou — APPROVED** [review]

**Review by jiri-one — APPROVED** [review]

**Pull request merged**